src/docfx/lib/markdown/EnsureParentExtension.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Markdig;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+
+namespace Microsoft.Docs.Build
+{
+    /// <summary>
+    /// Allow trace back to root node for any markdown object: https://github.com/lunet-io/markdig/issues/461
+    /// </summary>
+    internal static class EnsureParentExtension
+    {
+        private static readonly object s_parentKey = new object();
+
+        public static MarkdownPipelineBuilder UseEnsureParent(this MarkdownPipelineBuilder builder)
```

Comment:
can we add some tests on this extension?

Reply:
The effect of this extension `GetPathToRoot` used by `GetFilePath` in this PR has been exercised thoroughly. (see the 100% code coverage on `EnsureParentExtension.cs`). We generally don't write UT in docfx on specific methods, especially when it ties to some framework.